R/jhu_data.R

```diff
@@ -283,3 +251,74 @@ load_jhu_data <- function(
 
   return(results)
 }
+
+
+#' Create a tibble of incident and cumulative deaths or cases based on a
+#' specified issue date and measure. This function will also download a
+#' specified time series data set if issue_date is not available in pre-built
+#' data object.
+#' 
+#' @param issue_date character issue date (i.e. report date) to use for
+#' constructing truths in format 'yyyy-mm-dd'
+#' @param as_of character issue date (i.e. report date) to use for
+#' constructing truths published as of this date in format 'yyyy-mm-dd'
+#' @param measure character vector specifying measure of covid prevalence:
+#' 'deaths' or 'cases'
+#' 
+#' @return tibble with issue_date and data
+#' 
+preprocess_jhu_data <- function(issue_date = NULL, as_of = NULL, measure = "deaths"){
+  
+  if (measure == "deaths"){
+    links <- covidData::jhu_deaths_data_links
+    jhu_data <- covidData::jhu_deaths_data
+    base_file_name <- "time_series_covid19_deaths_US.csv"
+  } else if (measure == "cases"){
+    links <- covidData::jhu_cases_data_links
+    jhu_data <- covidData::jhu_cases_data
+    base_file_name <- "time_series_covid19_confirmed_US.csv"
+  }
+  
+  # validate issue_date and as_of
+  if (!missing(issue_date) && !missing(as_of) &&
+      !is.null(issue_date) && !is.null(as_of)) {
+    stop("Cannot provide both arguments issue_date and as_of to load_jhu_data.")
+  } else if (is.null(issue_date) && is.null(as_of)) {
+    issue_date <- max(jhu_data$issue_date)
+  } else if (!is.null(as_of)) {
+    avail_issues <- jhu_data$issue_date[
+      jhu_data$issue_date <= as.character(as_of)
+    ]
+    
+    if (length(avail_issues) == 0) {
+      stop("Provided as_of date is earlier than all available issue dates.")
+    } else {
+      issue_date <- max(avail_issues)
+    }
+  } else {
+    issue_date <- as.character(lubridate::ymd(issue_date))
+  }
+  
+  #issue_date <- as.Date(issue_date)
+  
+  # subset jhu_data based on issue_date
+  if (issue_date %in% jhu_data$issue_date){
+    jhu_data <- jhu_data %>%
+      dplyr::filter(issue_date == UQ(issue_date))
+  } else {
+    if(!issue_date %in% links$date){
```

Comment:
In one instance the condition in this if block evaluated to `FALSE` because the issue date was cast to a character on line 299 above, but the class of `links$date` was `Date`.  The command I ran was:

`covidData::load_jhu_data(issue_date = "2020-04-10", temporal_resolution="daily")`